packages/core/src/rules/function_module_recommendations.ts

```diff
@@ -0,0 +1,89 @@
+import {ABAPFile, Issue, Version} from "..";
+import {FunctionName} from "../abap/2_statements/expressions";
+import {ABAPRule} from "./_abap_rule";
+import {BasicRuleConfig} from "./_basic_rule_config";
+import {IRuleMetadata, RuleTag} from "./_irule";
+
+
+export type Recommendations = {name: string, replace: string, from?: Version;};
+
+export class FunctionModuleRecommendationsConf extends BasicRuleConfig {
+
+  /** Tuple of Function Module Name to be replaced, the recommended alternative and
+   * the version from which the recommendation is valid.*/
+  public recommendations: Recommendations[] = [
+    {name: "SUBST_GET_FILE_LIST", replace: "see note 1686357"},
+    {name: "ROUND", replace: "use built in function: round()"},
+    {name: "SO_NEW_DOCUMENT_ATT_SEND_API1", replace: "use CL_BCS"},
+    {name: "ECATT_CONV_XSTRING_TO_STRING", replace: "use CL_BINARY_CONVERT"},
+    {name: "SCMS_STRING_TO_XSTRING", replace: "use CL_BINARY_CONVERT"},
+    {name: "JOB_CREATE", replace: "use CL_BP_ABAP_JOB"},
+    {name: "JOB_SUBMIT", replace: "use CL_BP_ABAP_JOB"},
+    {name: "GUI_DOWNLOAD", replace: "use CL_GUI_FRONTEND_SERVICES"},
+    {name: "GUI_UPLOAD", replace: "use CL_GUI_FRONTEND_SERVICES"},
+    {name: "WS_FILENAME_GET", replace: "use CL_GUI_FRONTEND_SERVICES"},
+    {name: "F4_FILENAME", replace: "use CL_GUI_FRONTEND_SERVICES"},
+    {name: "SAPGUI_PROGRESS_INDICATOR", replace: "use CL_PROGRESS_INDICATOR"},
+    {name: "GUID_CREATE", replace: "use CL_SYSTEM_UUID"},
+    {name: "SSFC_BASE64_DECODE", replace: "use class CL_HTTP_UTILITY methods"},
+    {name: "SSFC_BASE64_ENCODE", replace: "use class CL_HTTP_UTILITY methods"},
+    {name: "SCMS_BASE64_DECODE_STR", replace: "use class CL_HTTP_UTILITY methods"},
+    {name: "POPUP_TO_DECIDE", replace: "use POPUP_TO_CONFIRM"},
+    {name: "REUSE_ALV_GRID_DISPLAY", replace: "use CL_SALV_TABLE=>FACTORY or CL_GUI_ALV_GRID"},
+    {name: "CALCULATE_HASH_FOR_RAW", replace: "use CL_ABAP_HMAC"},
+  ];
+}
+
+export class FunctionModuleRecommendations extends ABAPRule {
+
+  private conf = new FunctionModuleRecommendationsConf();
+
+  public getMetadata(): IRuleMetadata {
+    return {
+      key: "function_module_recommendations",
+      title: "Function Module Recommendations",
+      shortDescription: `Function Module Recommendations`,
+      extendedInformation: `https://docs.abapopenchecks.org/checks/53/`,
+      tags: [RuleTag.Styleguide],
```

Comment:
If its part of the ABAP style guide somewhere, please add a link under extendedInformation